cl/search/admin.py

```diff
@@ -186,6 +187,11 @@ def seal_documents(self, request: HttpRequest, queryset: QuerySet) -> None:
             [rd.pk for rd in queryset], "search.RECAPDocument"
         )
 
+        # Do a CloudFront invalidation
+        invalidate_cloudfront(
+            [f"/{rd.filepath_local.name}" for rd in queryset]
+        )
```

Comment:
```suggestion
# Do a CloudFront invalidation
invalidate_cloudfront(
            [f"/{rd.filepath_local.name}"]
)
```

Seems that it would be enough to pass `[f"/{rd.filepath_local.name}"]` to `invalidate_cloudfront` since the method is being called inside the `queryset` for loop and doing `[f"/{rd.filepath_local.name}" for rd in queryset] ` will send the paths to invalidate multiple times due to the nested loop.

Also, this code should be called before doing `rd.filepath_local.delete()` otherwise when sending the path to invalidate it will be empty. Alternatively, the `filepath_local.name` could be stored in a variable before deleting the file so that it can be passed then in` invalidate_cloudfront`.

With these two changes I think it should work well, I tried to test the invalidation but it doesn't work in my dev environment I think due to `CloudFront` permissions (I got a `NoCredentialsError`)

Reply:
Thanks, good catch. I just pushed a slightly different approach, which is to cache the file paths from the thumbnails and the files, and then do a single invalidation with those. That fixes another issue I noticed (we weren't invalidating thumbnails in CloudFront), and it makes it possible to only send one invalidation request even if we're sealing a dozen files. 

I've never actually sealed more than one file at a time using the admin tools, but I didn't like putting the invalidation in the queryset loop because it's more efficient to just have the one invalidation for all the files in the queryset, instead of having one *per* file in the queryset.